qiskit_experiments/calibration_management/save_utils.py

```diff
@@ -0,0 +1,253 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A helper module to save calibration data to local storage.
+
+.. warning::
+
+    This module is expected to be internal and no stable user-facing API
+    may not be guaranteed.
+
+.. note::
+
+    These model version must be upgraded when :class:`.Calibrations` data structure changes
+    in future and when the current data model becames unable to support it.
+
+"""
+
+from dataclasses import dataclass, field, asdict
+from datetime import datetime
+from typing import List, Dict, Any
+
+from qiskit.pulse import ScheduleBlock
+
+from .calibrations import Calibrations
+from .control_channel_map import ControlChannelMap
+from .parameter_value import ParameterValue
+
+
+@dataclass(frozen=True)
+class ParameterModelV1:
+    """A data schema of a single calibrated parameter.
+
+    .. note::
+        This is intentionally agnostic to the data structure of
+        Qiskit Experiments Calibrations for portability.
+
+    """
+
+    param_name: str
+    qubits: List[int]
+    schedule: str = ""
+    value: float = 0.0
+    datetime: datetime = datetime.now()
+    valid: bool = True
+    exp_id: str = ""
+    group: str = ""
+
+    @classmethod
+    def apply_schema(cls, data: Dict[str, Any]):
+        """Consume dictionary and returns canonical data model."""
+        return ParameterModelV1(**data)
+
+
+@dataclass(frozen=True)
+class ScheduleModelV1:
+    """A data schema of a single templated schedule.
```

Comment:
```suggestion
    """A data schema of a single template schedule.
```

Reply:
This is needed because schedule does not have any qubit information right? But Calibrations adds this info (number of qubits or specific qubits). Can the docstring state this?

Reply:
Your comment makes me think we should tie qubit information to the schedule itself, rather than respecting the current Calibration model. I removed this schema and attached them to schedule metadata in the formatted data domain.

Reply:
Should this be a separate issue/PR. Seems a bit tangential to the fact of serializing the calibrations.